pac4j-core/src/main/java/org/pac4j/core/client/DirectClientV2.java

```diff
@@ -38,7 +39,14 @@ protected C retrieveCredentials(final WebContext context) throws HttpAction {
             this.authenticator.validate(credentials, context);
             return credentials;
         } catch (CredentialsException e) {
-            logger.error("Failed to retrieve or validate credentials", e);
+            if (logger.isDebugEnabled()) {
```

Comment:
Nop, it's on purpose, because I don't want it to be a debug statement!

Another option would have been to have the info statement and then add a debug statement with the exception, but I don't like to have two log statement for the same information, so I usually change what is exactly logged (here adding the whole stacktrace instead of the message) depending on the configured level.